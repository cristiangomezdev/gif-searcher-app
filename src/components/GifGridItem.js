import React from "react";
import PropTypes from 'prop-types';
export const GifGridItem = ({id,title,url})=>{
    return (
        <div className="card animate__animated animate__fadeIn animate__delay-0s"><img src={url} alt={title} key={id} ></img><p>{title}</p></div>)
}

GifGridItem.propTypes={
    id:PropTypes.string.isRequired,
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired,
 }