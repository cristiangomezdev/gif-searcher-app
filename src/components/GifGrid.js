import React from "react";
import { GifGridItem } from "./GifGridItem";
//import { PropTypes } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
export const GifGrid = ({category})=>{
    const {data: images,loading} = useFetchGifs(category);

    return (<React.Fragment>
        <h3 className="animate__animated animate__fadeIn animate__delay-3s">{category}</h3>
        {loading && <p className="animate__animated animate__flash">Loading</p> }
        { <div className="card-grid">
        
            {images.map(image => <GifGridItem key={image.id} id={image.id} title={image.title} url={image.url} />)}
        </div> }
        </React.Fragment>)
}
//GifGrid.propTypes = { category: PropTypes.string.isRequired}