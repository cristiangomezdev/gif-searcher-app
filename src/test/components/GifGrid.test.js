import React from 'react';
import {shallow} from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('pruebas GifGrid', () => {
    const category = "one piece";

    const gifs =[{
        id:'ABC',url:'https://localhost/',title:'Cualquier cosa'},
        {id:'ABC',url:'https://localhost/',title:'Cualquier cosa'}]
    beforeEach(()=>{
        jest.clearAllMocks();

    })
    test('debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading:false
        });
        const wrapper = shallow(<GifGrid category={category} />)
        expect (wrapper).toMatchSnapshot();
    });
    test('debe de mostrar items cuando se cargan imagenes usefetchgifs',()=>{

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });
  
        const wrapper = shallow(<GifGrid category={category} />)

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
})  