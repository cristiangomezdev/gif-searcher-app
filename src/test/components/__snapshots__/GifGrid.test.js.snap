// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas GifGrid debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-3s"
  >
    one piece
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
